src/ioctls/vm.rs

```diff
@@ -418,36 +430,61 @@ impl VmFd {
     /// };
     /// unsafe { vm.set_user_memory_region(mem_region).unwrap() };
     ///
-    /// // Dummy x86 code that just calls halt.
-    /// let x86_code = [
-    ///         0xf4,             /* hlt */
+    /// #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+    /// // ASM code that just forces a MMIO Write.
+    /// let asm_code = [
+    ///         0xc6, 0x06, 0x00, 0x80, 0x00,
+    /// ];
+    /// #[cfg(target_arch = "aarch64")]
+    /// let asm_code = [
+    ///     0x01, 0x00, 0x00, 0x10, /* adr x1, <this address> */
+    ///     0x22, 0x10, 0x00, 0xb9, /* str w2, [x1, #16]; write to this page */
```

Comment:
Probably this is a stupid question, but I don't understand why you need this write. On x86_64 loading the code will dirty one page, on aarch64 it seems that this doesn't happen. Is this write required to dirty one page?

Reply:
Indeed, on aarch64 no dirty bit is set when the code is loaded into memory so yes, the write is required to dirty one page. 
After a short research it might seem that there are some specific dirty pages tracking on x86 (like for example [these](https://elixir.free-electrons.com/linux/v4.14.162/source/arch/x86/include/asm/kvm_host.h#L1040)), so it might be that those make a difference.